inc/Parsing/Parser.php

```diff
@@ -107,7 +107,21 @@ public function parse($doc) {
         // Normalize CRs and pad doc
         $doc = "\n" . str_replace("\r\n", "\n", $doc) . "\n";
         $this->lexer->parse($doc);
-        $this->handler->finalize();
+
+        if (!method_exists($this->handler, 'finalize')) {
+            /** @deprecated 2019-10 we have a legacy handler from a plugin, assume legacy _finalize exists */
+
+            \dokuwiki\Debug\DebugHelper::dbgCustomDeprecationEvent(
+                'finalize()',
+                get_class($this->handler) . '::_finalize()',
+                __METHOD__,
+                __FILE__,
+                __LINE__
+            );
+            $this->handler->_finalize();
```

Comment:
> The error reported by Scrutinizer is exactly the point of this pull request, but I might be able to add a Scrutinizer-disable comment if that is desired?

I would say add it, like a deprecation mark.

Reply:
Meh. It appears I was overly optimistic based due to my experience with the static analyzer phpstan, but apparently it is _not_ possible to tell Scrutinizer to ignore a rule for a line with a comment.